clang/lib/Sema/SemaOpenMP.cpp

```diff
@@ -5965,6 +5967,244 @@ static bool teamsLoopCanBeParallelFor(Stmt *AStmt, Sema &SemaRef) {
   return Checker.teamsLoopCanBeParallelFor();
 }
 
+static Expr *getInitialExprFromCapturedExpr(Expr *Cond) {
+
+  Expr *SubExpr = Cond->IgnoreParenImpCasts();
+
+  if (auto *DeclRef = dyn_cast<DeclRefExpr>(SubExpr)) {
+    if (auto *CapturedExprDecl =
+            dyn_cast<OMPCapturedExprDecl>(DeclRef->getDecl())) {
+
+      // Retrieve the initial expression from the captured expression
+      return CapturedExprDecl->getInit();
+    }
+  }
+  return nullptr;
+}
+
+static Expr *copy_removePseudoObjectExpr(const ASTContext &Context, Expr *E,
+                                         SemaOpenMP *SemaPtr, bool NoContext) {
+
+  BinaryOperator *BinaryCopyOpr = NULL;
+  bool BinaryOp = false;
+  if (E->getStmtClass() == Stmt::BinaryOperatorClass) {
+    BinaryOp = true;
+    BinaryOperator *E_BinOpr = static_cast<BinaryOperator *>(E);
+    BinaryCopyOpr = BinaryOperator::Create(
+        Context, E_BinOpr->getLHS(), E_BinOpr->getRHS(), E_BinOpr->getOpcode(),
+        E_BinOpr->getType(), E_BinOpr->getValueKind(),
+        E_BinOpr->getObjectKind(), E_BinOpr->getOperatorLoc(),
+        FPOptionsOverride());
+    E = BinaryCopyOpr->getRHS();
+  }
+
+  // Change PseudoObjectExpr to a direct call
+  if (PseudoObjectExpr *PO = dyn_cast<PseudoObjectExpr>(E))
+    E = *((PO->semantics_begin()) - 1);
```

Comment:
Why do you need this?

Reply:
I have added comments at the beginning of the function CloneAssociatedStmt(). I think this answers your question. Please let me know if I have to add comments to this statement in particular.